taskcluster/scripts/generate-screenshots.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import asyncio
+import logging
+import sys
+
+from aiohttp_retry import RetryClient
+
+
+log = logging.getLogger(__name__)
+
+BITRISE_APP_SLUG_ID = "6c06d3a40422d10f"
+BITRISE_URL_TEMPLATE = "https://api.bitrise.io/v0.1/apps/" + BITRISE_APP_SLUG_ID + "/{suffix}"
+
+
+class TaskException(Exception):
+    def __init__(self, *args, exit_code=1):
+        """Initialize ScriptWorkerTaskException.
+        Args:
+            *args: These are passed on via super().
+            exit_code (int, optional): The exit_code we should exit with when
+                this exception is raised.  Defaults to 1 (failure).
+        """
+        self.exit_code = exit_code
+        super(Exception, self).__init__(*args)
+
+
+def sync_main(
+    loop_function=asyncio.get_event_loop,
+):
+    """
+    This function sets up the basic needs for a script to run. More specifically:
+        * it initializes the logging
+        * it creates the asyncio event loop so that `async_main` can run
+    Args:
+        loop_function (function, optional): the function to call to get the
+            event loop; here for testing purposes. Defaults to
+            ``asyncio.get_event_loop``.
+    """
+    _init_logging()
+
+    parser = argparse.ArgumentParser(description="Generate a screenshot by delegating the work to bitrise.io")
+
+    parser.add_argument("--token-file", required=True, type=argparse.FileType("r"), help="file that contains the bitrise.io token")
+    parser.add_argument("--branch", required=True, help="the git branch to generate screenshots from")
+    parser.add_argument("--commit", required=True, help="the git commit hash to generate screenshots from")
+    parser.add_argument("--locale", required=True, help="locale to generate the screenshots for")
+
+    result = parser.parse_args()
+
+    token = result.token_file.read().strip()
+    if token.rstrip() == "faketoken":
+        log.warn('"faketoken" detected. Not uploading anything to the service.')
+        sys.exit(0)
+
+    loop = loop_function()
+    loop.run_until_complete(_handle_asyncio_loop(
```

Comment:
If you're on 3.7 or higher, `asyncio.run` is the new standard way to do this. It handles all of the loop mechanics internally.

Reply:
Niiiiiice, TIL! 

At the moment, the docker image install Python 3.6: https://firefox-ci-tc.services.mozilla.com/tasks/BIanatygRrSnCOTanEDeUg/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FBIanatygRrSnCOTanEDeUg%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log#L382

I'll keep that in mind for the next upgrade!